src/stcal/resample/utils.py

```diff
@@ -423,3 +422,29 @@ def is_flux_density(bunit):
     except (ValueError, TypeError):
         flux_density = False
     return flux_density
+
+
+def _get_inverse_variance(array, data_shape):
+    """
+    Compute the inverse variance array for weighting.
+
+    Parameters
+    ----------
+    array : numpy.ndarray or None
+        Input variance array (e.g., VAR_RNOISE or VAR_SKY).
+    data_shape : tuple
+        Expected shape of the output array.
+
+    Returns
+    -------
+    inv : numpy.ndarray
+        Inverse variance array. If input array is missing or has wrong shape,
+        returns an array filled with zeros of shape `data_shape`.
+    """
+    if array is not None and array.shape == data_shape:
+        with np.errstate(divide="ignore", invalid="ignore"):
+            inv = 1.0 / array
+        inv[~np.isfinite(inv)] = 0
+    else:
+        inv = np.full(data_shape, 0, dtype=np.float32)
```

Comment:
```suggestion
        inv = np.full(data_shape, 1, dtype=np.float32)
```
Let's retain the old behavior for this failure case (we asked for some weighting that didn't exist and we just provide all 1).

Reply:
Done.